core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterSampleTest.java

```diff
@@ -0,0 +1,31 @@
+package test.failedreporter.issue2517;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+public class FailedDataProviderWithFactoryReporterSampleTest {
+  private Integer data;
+
+  public FailedDataProviderWithFactoryReporterSampleTest() {
```

Comment:
It should work without the empty constructor. Could you remove it if it is working well without it?

Reply:
![image](https://user-images.githubusercontent.com/60096114/114395361-ef7fbd00-9ba4-11eb-8456-066e540a9f10.png)